sharding/shard_test.go

```diff
@@ -0,0 +1,68 @@
+package sharding
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+)
+
+func TestShard_ValidateShardID(t *testing.T) {
+	header := &CollationHeader{shardID: big.NewInt(4)}
+	shard := MakeShard(big.NewInt(3))
+
+	if err := shard.ValidateShardID(header); err == nil {
+		t.Fatalf("Shard ID validation incorrect. Function should throw error when shardID's do not match. want=%d. got=%d", header.ShardID().Int64(), shard.ShardID().Int64())
```

Comment:
This should be Errorf. You should only fatal when the test cannot possibly proceed forward. 

Please change this everywhere in this test